### ISSUE TITLE
changefeedccl: data race in test helpers

### ISSUE THREAD

**Issue by jayshrivastava** [cockroachdb/cockroach#109966]
Seen here https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelExtendedCi/11604668?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildChangesSection=true&expandBuildProblemsSection=true&expandBuildTestsSection=true&showLog=11604666_1943_208.1560.1941&logFilter=debug&logView=flowAware

```
 WARNING: DATA RACE
      Write at 0x00c005d1127c by goroutine 376235:
        github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.makeOptions.func1.1()
            github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/helpers_test.go:659 +0x37
        github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/kvfeed.(*rangefeed).addEventsToBuffer()
            github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/kvfeed/physical_kv_feed.go:123 +0x3f0
        github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/kvfeed.(*rangefeed).addEventsToBuffer-fm()
            <autogenerated>:1 +0x4d
        github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1()
            github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:168 +0x4c
        golang.org/x/sync/errgroup.(*Group).Go.func1()
            golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:75 +0x86
      
      Previous read at 0x00c005d1127c by goroutine 375230:
        github.com/cockroachdb/cockroach/pkg/util/hlc.(*Timestamp).Size()
            github.com/cockroachdb/cockroach/pkg/util/hlc/bazel-out/k8-fastbuild/bin/pkg/util/hlc/hlc_go_proto_/github.com/cockroachdb/cockroach/pkg/util/hlc/timestamp.pb.go:307 +0x204
        github.com/cockroachdb/cockroach/pkg/kv/kvpb.(*RangeFeedCheckpoint).Size()
            github.com/cockroachdb/cockroach/pkg/kv/kvpb/bazel-out/k8-fastbuild/bin/pkg/kv/kvpb/kvpb_go_proto_/github.com/cockroachdb/cockroach/pkg/kv/kvpb/api.pb.go:23934 +0x217
        github.com/cockroachdb/cockroach/pkg/kv/kvpb.(*RangeFeedEvent).Size()
            github.com/cockroachdb/cockroach/pkg/kv/kvpb/bazel-out/k8-fastbuild/bin/pkg/kv/kvpb/kvpb_go_proto_/github.com/cockroachdb/cockroach/pkg/kv/kvpb/api.pb.go:23991 +0xd2
        github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/kvevent.(*Event).ApproximateSize()
            github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/kvevent/event.go:128 +0x146
        github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/kvevent.(*throttlingBuffer).Get()
            github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/kvevent/throttling_buffer.go:35 +0x12b
        github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*changeAggregator).tick()
            github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_processors.go:653 +0x116
        github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*changeAggregator).Next()
            github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_processors.go:615 +0x19a
        github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*changeFrontier).Next()
            github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_processors.go:1274 +0x1fa
        github.com/cockroachdb/cockroach/pkg/sql/execinfra.Run()
            github.com/cockroachdb/cockroach/pkg/sql/execinfra/base.go:196 +0x59
        github.com/cockroachdb/cockroach/pkg/sql/execinfra.(*ProcessorBaseNoHelper).Run()
            github.com/cockroachdb/cockroach/pkg/sql/execinfra/processorsbase.go:726 +0xcf
        github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*changeFrontier).Run()
            <autogenerated>:1 +0x6b
        github.com/cockroachdb/cockroach/pkg/sql/flowinfra.(*FlowBase).Run()
            github.com/cockroachdb/cockroach/pkg/sql/flowinfra/flow.go:574 +0x747
        github.com/cockroachdb/cockroach/pkg/sql/rowflow.(*rowBasedFlow).Run()
            <autogenerated>:1 +0x6e
        github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run()
            github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:907 +0x14d5
        github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.startDistChangefeed.func1()
            github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_dist.go:311 +0x6db
        github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1()
            github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:168 +0x4c
        golang.org/x/sync/errgroup.(*Group).Go.func1()
            golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:75 +0x86
      
      Goroutine 376235 (running) created at:
        golang.org/x/sync/errgroup.(*Group).Go()
            golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:72 +0x12e
        github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx()
            github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:167 +0x109
        github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/kvfeed.rangefeedFactory.Run()
            github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/kvfeed/physical_kv_feed.go:77 +0x335
        github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/kvfeed.(*kvFeed).runUntilTableEvent.func4()
            github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/kvfeed/kv_feed.go:510 +0x157
        github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1()
            github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:168 +0x4c
        golang.org/x/sync/errgroup.(*Group).Go.func1()
            golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:75 +0x86
      
      Goroutine 375230 (running) created at:
        golang.org/x/sync/errgroup.(*Group).Go()
            golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:72 +0x12e
        github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx()
            github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:167 +0x109
        github.com/cockroachdb/cockroach/pkg/util/ctxgroup.GoAndWait()
            github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:189 +0xb1
        github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.startDistChangefeed()
            github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_dist.go:315 +0x852
        github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.distChangefeedFlow()
            github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_dist.go:129 +0x5b4
        github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*changefeedResumer).resumeWithRetries()
            github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_stmt.go:1288 +0x5ea
        github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*changefeedResumer).Resume()
            github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_stmt.go:1123 +0x3ac
        github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*reportErrorResumer).Resume()
            github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/testfeed_test.go:343 +0x6f
        github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine.func2()
            github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1659 +0x1a2
        github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine()
            github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1660 +0xc51
        github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).runJob()
            github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:473 +0x851
        github.com/cockroachdb/cockroach/pkg/jobs.(*StartableJob).Start.func2()
            github.com/cockroachdb/cockroach/pkg/jobs/jobs.go:953 +0x252
        github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2()
            github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:484 +0x1f6
      ==================
```

Jira issue: CRDB-31192

**Comment by jayshrivastava** [comment]
Seen in this PR https://github.com/cockroachdb/cockroach/pull/109835. I believe the code changes in the PR are unrelated.

**Comment by jayshrivastava** [comment]
The write happens here https://github.com/cockroachdb/cockroach/blob/09c7f9f1569a432e719c077eebb119cc702fcd18/pkg/ccl/changefeedccl/helpers_test.go#L659

**Comment by jayshrivastava** [comment]
Happened again here https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelExtendedCi/11633881?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildChangesSection=true&expandBuildProblemsSection=true&expandBuildTestsSection=true&showLog=11633878_446_191.439.444&logFilter=debug&logView=flowAware

**Comment by miretskiy** [comment]
Fixed via #71221 110215
